zrml/prediction-markets/src/tests.rs

```diff
@@ -1125,6 +1125,160 @@ fn market_resolve_does_not_hold_liquidity_withdraw() {
     })
 }
 
+#[test]
+fn approve_market_correctly_unreserves_advisory_bond() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(PredictionMarkets::create_categorical_market(
+            Origin::signed(ALICE),
+            BOB,
+            MarketPeriod::Block(0..100),
+            gen_metadata(2),
+            MarketCreation::Advised,
+            2,
+            MarketDisputeMechanism::SimpleDisputes,
+            ScoringRule::CPMM,
+        ));
+        let alice_balance_before = Balances::free_balance(&ALICE);
+        assert_eq!(Balances::reserved_balance(&ALICE), AdvisoryBond::get() + OracleBond::get());
+        assert_ok!(PredictionMarkets::approve_market(Origin::signed(SUDO), 0));
+        assert_eq!(Balances::reserved_balance(&ALICE), OracleBond::get());
+        assert_eq!(Balances::free_balance(&ALICE), alice_balance_before + AdvisoryBond::get());
+    });
+}
+
+#[test]
+fn on_resolution_correctly_reserves_and_unreserves_bonds_for_permissionless_market_on_oracle_report()
+ {
+    ExtBuilder::default().build().execute_with(|| {
+        // We need to set the validity bond to something larger than the advisory bond in order to
+        // be able to detect if we're unreserving the correct amount!
+        <Runtime as Config>::ValidityBond::set(&(2 * AdvisoryBond::get()));
```

Comment:
Yeah, this current style of testing fits exactly the pattern as in `admin_destroy_market` tests, which I would have expected in the first place.

Reply:
I would see your point, that you'd like to check that the unreserve doesn't do too much than required. The fact with using `unreserve` is, that it will never get the `reserved_balance` below zero. So if a bond is larger than the `reserved_balance`, then you can not detect that it would actually unreserve more. So anyways, it doesn't change any major behaviour and I am okay with `SomeBond = 2 * AnotherBond`.